Documentation/learning/why.md

```diff
@@ -0,0 +1,21 @@
+# Why etcd
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
+
+# What does etcd stand for?
```

Comment:
remove this section. we probably should just mention it in the way part about its origin.